src/lib/chawathe98_matcher.rs

```diff
@@ -46,35 +46,2262 @@
+    pub fn delete_node(&mut self, node_to_delete: NodeId<FromNodeId>) {
+        self.list_node_deleted.borrow_mut().push(node_to_delete);
+    }
+    /// Print
+    pub fn print(&self) {
+        let size_from_node = NodeId::new(self.from_arena.borrow().size() - 1);
+        println!("{:?}", size_from_node);
+        let size_to_node = NodeId::<ToNodeId>::new(self.to_arena.borrow().size() - 1);
+        let last_node_from_node = &self.from_arena.borrow()[size_from_node];
+        println!("The Last node From_NODE is --> {:?}", last_node_from_node);
+        let last_node_to_node = &self.to_arena.borrow()[size_to_node];
+        println!("The Last node From_NODE is --> {:?}", last_node_to_node);
+        println!("List Of edges already mapped to");
+        let mut index = 0;
+        let list_edges = self.list_edges.borrow();
+        for edges in list_edges.iter() {
+            println!(" -- {:?} -- ", index);
+
+            println!(
+                "TY        :   From node    -> {:?} || to node -> {:?}",
+                self.from_arena.borrow()[edges.from_node].ty, self.to_arena.borrow()[edges.to_node].ty
+            );
+            println!(
+                "Value     :   From node -> {:?} || to node -> {:?}",
+                self.from_arena.borrow()[edges.from_node].label, self.to_arena.borrow()[edges.to_node].label
+            );
+            println!(
+                "From node -> {:?} || to node -> {:?} || Edge type -> {:?}",
+                edges.from_node, edges.to_node, edges.edge_type
+            );
+            index += 1;
+            println!("--");
+        }
+    }
+}
+
+/// The Chawathe98Config2 matcher needs no configuration.
 #[derive(Debug, Clone, PartialEq)]
-pub struct Chawathe98Config  {}
+pub struct Chawathe98Config2 {}
 
-impl Default for Chawathe98Config  {
-    fn default() -> Chawathe98Config  {
-        Chawathe98Config  {}
+impl Default for Chawathe98Config2 {
+    fn default() -> Chawathe98Config2 {
+        Chawathe98Config2 {}
     }
 }
 
-impl Chawathe98Config  {
+impl Chawathe98Config2 {
     /// Create a new configuration object, with default values.
-    pub fn new() -> Chawathe98Config  {
+    pub fn new() -> Chawathe98Config2 {
         Default::default()
     }
 }
 
-impl<T: Clone + Debug + Eq + 'static> MatchTrees<T> for Chawathe98Config  {
+impl<T: Clone + Debug + Eq + 'static> MatchTrees2<T> for Chawathe98Config2 {
+    /// Perform matches
+    fn match_trees(
+        &self,
+        base: Arena<T, FromNodeId>,
+        diff: Arena<T, ToNodeId>,
+    ) -> MappingStore2<T> {
+        let mut store = MappingStore2::new(base.clone(), diff.clone());
+        if store.from_arena.borrow().is_empty() || store.to_arena.borrow().is_empty() {
+            return store;
+        }
+
+        let last_node_from_node = NodeId::<FromNodeId>::new(base.size() - 1);
+        let last_node_to_node = NodeId::<ToNodeId>::new(diff.clone().size() - 1);
+        //let last_node_from_node = base[size_fromNode];
```

Comment:
In general, we don't want to merge commented out code into master, so let's kill these lines, and the ones below